docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
+* Support for expanding sparse VM definitions
+* Support for detailed VM definitions
+* Support hot-plugging
+* Representation of dynamic instance specific data
+* Support for defaulting (to support OS compatibility)
+Considerations
+* Sub-resources for actions
+* Presets to add certain configurations
+
+
+## Definitions
+* **Defaulting** - Setting certain defaults during expansion (i.e. virtio for a
+  bus type, or e1000 for as a nic model)
+* **Expansion** - VM definitions can be provided in a sparse way i.e. just a
+  disk. Once the VM is getting started, the runtime will infer additional
+  requirements for this device and add them to the runtime VM definition.
+  This process of inferring and adding additional devices is called expansion.
+* **Flavor** - Comparable to a template or a specific instance type
+* **Preset** - Presetting certain values in a specified path of the document
+  (i.e. setting .spec.domain.devices.nvram). It is not about setting certain
+  default values for undefined fields (i.e. set virtio for all busses)
```

Comment:
yes and no. Specifically we're not using the library in oVirt, but have a different database instead serving the same purpose. It's more of an extension really, because it contains a more fine grained definition of allowed devices and settings than libosinfo, and have differentiation for product version (i.e. Windows supports hotplug CPU, but oVirt supports that only in version 3.4+).
I'm not sure we can use libosinfo in the final project alone, I believe more details specific to that project would always be needed, but it is certainly the right thing to start with and then extend.

I would suggest to rethink defaulting...to me it also seems as something which is not really needed - if there is a 1:1 mapping between what you express in spec and the details(e.g. an expansion from distro suse-10 leading to addition of virtio-blk controller) and you want to express the derived items in the spec for customization(like the controller type, constrained by what libosinfo defines), then why do you need anything else than a preset?